frontend/templates/product/index.tsx

```diff
@@ -86,6 +84,13 @@ export const getServerSideProps: GetServerSideProps<ProductTemplateProps> =
          },
          handle
       );
+      const proOnly = product?.tags.find((tag: string) => tag === "Pro Only");
+      if (proOnly) {
+         context.res.setHeader('X-Robots-Tag', 'noindex, nofollow');
```

Comment:
This doesn't make it very easy to QA. Maybe we can noindex, follow on proOnly but noindex nofollow otherwise.